lib/readline.js

```diff
@@ -481,6 +482,42 @@ Interface.prototype._insertString = function(c) {
     // A hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+  // Emit current line for generating preview
+  this.emit('buffer', this.line);
+};
+
+// Append eager eval result to the
+// Current line
+Interface.prototype._appendPreview = function(result) {
+  this.previewResult = `\u001b[90m // ${result}\u001b[39m`;
+  const columns = this.output.columns;
+
+  // Cursor to left edge.
+  cursorTo(this.output, 0);
+  clearScreenDown(this.output);
+
+  if (columns) {
+    const colorDepth = this.output.getColorDepth();
+    const s = colorDepth > 256 ?
+      `${this._prompt}${this.line}${this.previewResult}` :
+      `${this._prompt}${this.line}${result}`;
+    this.output.write(s.length < columns ?
+      s : `${s.slice(0, columns - 3)
+        .replace(/\r?\n|\r/g, '')}...\u001b[39m`);
```

Comment:
This adds the color unconditionally. Can you also please elaborate what exactly this does? I am not sure I understand why this part is sliced and replaced.

Reply:
The slice and replace part is mainly for splitting the new lines. Since in output terminal has only limited columns, we need to split them and replace with empty space so that we can show them the content in single line.

Reply:
We already use `util.inspect()` on the preview. We should use options that deactivate the colors for the preview and make sure the output is printed on a single line. That can be done with the `compact` options set to `true` in combination with `breakLength` set to `Infinity` and `colors` set to `false`.

I would then limit the preview only instead of limiting the input as well. Right now we unconditionally slice the input value and that doesn't seem right.

Reply:
@BridgeAR: (sorry for a very late reply)

What you are saying is actually a good idea. But I was just trying to play with the options provided by you but still that doesn't seems to be atleast fixing the problem what we have now. For example, if the user types `process`, we get a long string, so if we apply the options like below:

```
util.inspect(process, {breakLength: Infinity, compact: true})
```

still we are getting results like below:

<img width="891" alt="Screen Shot 2019-11-27 at 11 36 47 AM" src="https://user-images.githubusercontent.com/1241511/69698046-3aad4300-110a-11ea-81e6-a4031143e271.png">

This actually breaks the cursor position as the output is huge. I feel, the only way for us to slice the string to the length of the tty `column`. May be is there any other way we could achieve it? 🤔